src/analysis/cache.Rmd

```diff
@@ -0,0 +1,313 @@
+axis(2)
+matplot(do.call(data.frame, split(t$context_switches, t$opcode)), type="l", xlab="op_count", ylab="context_switches",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$page_faults, t$opcode)), type="l", xlab="op_count", ylab="page_faults",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$instructions, t$opcode)), type="l", xlab="op_count", ylab="instructions",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$branches, t$opcode)), type="l", xlab="op_count", ylab="branches",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$branch_misses, t$opcode)), type="l", xlab="op_count", ylab="branch_misses",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$L1_dcache_loads, t$opcode)), type="l", xlab="op_count", ylab="L1_dcache_loads",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$LLC_loads, t$opcode)), type="l", xlab="op_count", ylab="LLC_loads",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$LLC_load_misses, t$opcode)), type="l", xlab="op_count", ylab="LLC_load_misses",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$L1_icache_loads, t$opcode)), type="l", xlab="op_count", ylab="L1_icache_loads",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$L1_icache_load_misses, t$opcode)), type="l", xlab="op_count", ylab="L1_icache_load_misses",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$dTLB_loads, t$opcode)), type="l", xlab="op_count", ylab="dTLB_loads",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$dTLB_load_misses, t$opcode)), type="l", xlab="op_count", ylab="dTLB_load_misses",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$iTLB_loads, t$opcode)), type="l", xlab="op_count", ylab="iTLB_loads",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+matplot(do.call(data.frame, split(t$iTLB_load_misses, t$opcode)), type="l", xlab="op_count", ylab="iTLB_load_misses",axes=FALSE)
+axis(side=1, at=1:11, labels=c(0,5,10,15,20,25,30,35,40,45,50))
+axis(2)
+```
+
+Looking at the graphs we can say that a few factors have linear dependencies. For the sake of completeness we calculate Pearson correlations.
```

Comment:
am I parsing this correctly? All of these variables are absolute measurements, so the linear dependencies tell us only that `perf` "sees" the impact of extending the EVM bytecode by adding instructions, correct?